fix: restore the use of file system storage

oetyng opened the pull request:
Resolves #211.

oetyng committed:
fix: restore the use of file system storage

joshuef commented:
I don't see any tests asserting that files are republished here. We'd need that if we're not relying on libp2p

oetyng committed:
refactor!: encaps peerid bytes instead of hashing

- This allows back and forth conversion which will be useful when
using libp2p::kad::KBucketKey closeness calcs.

oetyng committed:
feat: impl NetworkKey

- This will be useful for our needs to locate peers and data by a
common metric using the libp2p::kad::KBucketKey.

oetyng committed:
refactor!: impl the canonical network key

- This introduces a uniform and consistent usage of the KBucketKey logic
for distance calculation, whereby our own XorName calcs can and shall be
discontinued.

oetyng committed:
feat: impl get missing data query

- First step of the data replication feature, which allows nodes to send
the data addresses they have (for the moment chunks only) to the close
group, as to have them send the data that they should hold but do not.

oetyng committed:
feat: impl ask for missing data on peer died

- This ties together the data replication flow, by pageing close nodes
for the missing data.

oetyng committed:
chore: kill and start new node for churn tests

- If we just restart, all data will still be there, so replication won't
really be tested.

oetyng committed:
chore: temporarily run churn on PR

oetyng committed:
chore: skip ghost node

- Some double of one of the nodes gets added for some strange
reason, always having pid 0. Skipping it until the cause can be found.

oetyng committed:
chore: only run the churn tasks on ubuntu

oetyng committed:
chore: improve some logging

oetyng committed:
chore: remove peer instead of addr from kad

- Temp test to see if that is what we actually want.
